cmd/service-controller/bridges.go

```diff
@@ -216,9 +217,14 @@ func (c *Controller) updateServiceBindings(required types.ServiceInterface, port
 		} else if bindings.headless != nil {
 			bindings.headless = nil
 		}
+
+		hasSkupperSelector := false
 		for _, t := range required.Targets {
 			targetPort := getTargetPort(required, t)
-			if t.Selector != "" {
+			if strings.Contains(t.Selector, "skupper.io/component=router") {
```

Comment:
I don't think the target definition should ever have `skupper.io/component=router` in it. I.e. I suspect that we want to prevent that happening in the first place?

Reply:
Well, that is not what is currently happening. If you use latest image from quay (0.3),
the selector isn't changing, as you say. But if you build a fresh image from master,
then the selector is changing when you add `skupper.io/proxy` to an existing service
without specifying `skupper.io/address` (it is the actual initial reason for #299).

Reply:
I think this check - or an equivalent - should be happening in the definition_monitor.go, to prevent the service definition having the wrong selector written in to it in the first place.

I.e. in getServiceDefinitionFromAnnotatedService, where the selector is not nil and the target annotation is not set, we should check whether the selector is already the router selector, if not that is the selector to use. If it is already the router selector then we should use the original selector annotation value instead. (I think!)

Reply:
Gotcha! It makes sense to me (if I understood correctly). Let me implement it and I
will let you know once code has been updated and tested locally.

Reply:
Thanks Fernando! Sorry for the chopping and changing. I had not properly appreciated the complexities in this case.

Reply:
Come on, I am the one that have to thank you for your, always, valuable advices :+1: